Sources/PaylikeSDK/Views/ViewModel/SimpleWhitelabelPaymentFormViewModel.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  SimpleWhitelabelPaymentFormViewModel.swift
+//  
+//
+//  Created by Székely Károly on 2023. 05. 19..
+//
+
+import Foundation
+import PaylikeEngine
+import PaylikeClient
+
+typealias onSuccessHandler = () -> Void
+typealias onErrorHandler = () -> Void
+
+public class SimpleWhitelabelPaymentFormViewModel: ObservableObject {
+    private var engine: PaylikeEngine
+    
+    private var onSuccess: onSuccessHandler
+    private var onError: onErrorHandler
+    
+    @Published var cardNumber: String = "";
```

Comment:
Are the implicit internal accessibility modifiers good for these published variables?

They are used in the [SimpleWhitelabelPaymentForm.swift](https://github.com/kocsislaci/swift-sdk/pull/2/files#diff-17febfcf75b6b2cb8c42a6fba105c6ab9ffda492a398b87c01aa40d10ec88fdc) along with the validator functions (e.g. `isCardNumberValid`), but they are public.

They are used at the same level but have different accessibility :)

Reply:
as the validators are internal functions, and the setter is needed by the view, I think at the moment its fine to leave them as internal. I have made the viewModel private in SimpleWhitelabelPaymentForm.swift, but I'm not sure I understand what would be a proper access level inside the viewModel, so I would be glad to hear your opinion! :)

Reply:
Inside the `SimpleWhitelabelPaymentForm`, the private viewModel is good, I believe.

The internal accessibility for the published fields is also good. On the contrary, the `isCardNumberValid` and other "isValid" functions are public.
Should not they (states and isValid func-s) be at the same level? Because they are used in the same line in the view (`SimpleWhitelabelPaymentForm`).

`CardNumberField(cardNumber: $viewModel.cardNumber, isValid: viewModel.isCardNumberValid)`

I should have addressed the question to the public `isValid` functions in the first place :D

This level depends on the design we provide for the users. It can be public if we want to expose the API to enable a third party to create a custom view for the view model; otherwise, internal is enough.

I think internal is good for now, later on it can be changed.

Reply:
Now i get it, I will take a harder look at the public interface of this viewModel tomorrow, and take this into consideration! :)